File: backend/src/main/java/net/pengcook/recipe/dto/AuthorResponse.java
package net.pengcook.recipe.dto;

import net.pengcook.user.domain.User;

public record AuthorResponse(long authorId, String authorName, String authorImage) {

    public AuthorResponse(User author) {
        this(author.getId(), author.getUsername(), author.getImage());
    }
}

File: android/app/src/main/java/net/pengcook/android/data/model/feed/item/AuthorResponse.kt
package net.pengcook.android.data.model.feed.item


import com.google.gson.annotations.SerializedName

data class AuthorResponse(
    @SerializedName("authorId")
    val authorId: Long,
    @SerializedName("authorImage")
    val authorImage: String,
    @SerializedName("authorName")
    val authorName: String
)
